File: packages/gem_core/health.py
from __future__ import annotations

from dataclasses import dataclass
from time import time


STARTED_AT = time()


@dataclass(frozen=True)
class Health:
    service: str
    status: str = "ok"
    version: str = "0.1.0"

    def as_dict(self) -> dict[str, str | float]:
        return {
            "service": self.service,
            "status": self.status,
            "version": self.version,
            "uptime_seconds": round(time() - STARTED_AT, 3),
        }

File: src/main/kotlin/com/onexeor/gemos/core/Health.kt
package com.onexeor.gemos.core

import kotlinx.serialization.Serializable
import kotlin.time.DurationUnit
import kotlin.time.TimeSource

private val startedAt = TimeSource.Monotonic.markNow()

@Serializable
data class HealthResponse(
    val service: String,
    val status: String = "ok",
    val version: String = "0.1.0",
    val uptimeSeconds: Double = startedAt.elapsedNow().toDouble(DurationUnit.SECONDS),
)

